cpp/src/io/parquet/parquet_gpu.hpp

```diff
@@ -83,6 +83,35 @@ enum level_type {
   NUM_LEVEL_TYPES
 };
 
+/**
+ * @brief Nesting information specifically needed by the decode and preprocessing
+ * kernels.
+ *
+ * This data is kept separate from PageNestingInfo to keep it as small as possible.
+ * It is used in a cached form in shared memory when possible.
+ */
+struct PageNestingDecodeInfo {
+  // set up prior to decoding
+  int32_t max_def_level;
```

Comment:
Verifying our intended spellings: Should all these types by `int32_t` , or should some be `int` or `cudf::size_type`?

Reply:
How far do we want to take this here?  Broadly, cuio tends to work with native types in the kernels and not cudf typedefs (mostly for historical reasons, not really philosophically).  I think my preference here would be to keep `int32_t` for now (although there's some errant `int` fields worth fixing up) and then maybe file an issue to do a more detailed audit cleanup.